src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Completion/SqlCompletionItem.cs

```diff
@@ -48,14 +49,38 @@ public SqlCompletionItem(string declarationTitle, DeclarationType declarationTyp
         private void Init()
         {
             InsertText = DeclarationTitle;
-            DelimitedIdentifier delimitedIdentifier = GetDelimitedIdentifier(TokenText);
             Label = DeclarationTitle;
+            DelimitedIdentifier delimitedIdentifier = GetDelimitedIdentifier(TokenText);
 
-            if (delimitedIdentifier == null && !string.IsNullOrEmpty(DeclarationTitle) && 
-                (!ValidSqlNameRegex.IsMatch(DeclarationTitle) || AutoCompleteHelper.IsReservedWord(InsertText)))
+            // If we're not already going to quote this then handle special cases for various
+            // DeclarationTypes
+            if (delimitedIdentifier == null && !string.IsNullOrEmpty(DeclarationTitle))
             {
-                InsertText = WithDelimitedIdentifier(BracketedIdentifiers, DeclarationTitle);
+                switch (this.DeclarationType)
+                {
+                    case DeclarationType.Server:
+                    case DeclarationType.Database:
+                    case DeclarationType.Table:
+                    case DeclarationType.Column:
+                    case DeclarationType.View:
+                    case DeclarationType.Schema:
+                        // Only quote if we need to - i.e. if this isn't a valid name (has characters that need escaping such as [) 
+                        // or if it's a reserved word
+                        if (!ValidSqlNameRegex.IsMatch(DeclarationTitle) || AutoCompleteHelper.IsReservedWord(InsertText)) {
+                            InsertText = WithDelimitedIdentifier(BracketedIdentifiers, DeclarationTitle);
+                        }
+                        break;
+                    case DeclarationType.BuiltInFunction:
+                    case DeclarationType.ScalarValuedFunction:
+                    case DeclarationType.TableValuedFunction:
```

Comment:
should there be a `default` case?

Reply:
We already default InsertText to be the DeclarationTitle above so having a separate default here isn't necessary I think